plugins/woocommerce/client/admin/client/payments/payment-recommendations-wrapper.tsx

```diff
@@ -24,7 +24,8 @@ export const PaymentRecommendations: React.FC< EmbeddedBodyProps > = ( {
 	if (
 		page === 'wc-settings' &&
 		tab === 'checkout' &&
-		( ! section || section === 'main' )
+		( ! section || section === 'main' ) &&
```

Comment:
Nitpick: Is the component visibility logic not better suited in its container?

Reply:
sorry, I didn't get this one. What do you mean?

Reply:
So, right now the component is deciding whether to show or hide itself. I think the component should simply care about its markup/logic and not its own visibility. That's better suited in the container component where it will be used.

Having said that, it's just a convention I've seen used around and not a best practice or standard.

Reply:
oh, I see! so, it's basically the wrapper that handles this logic. See the `PaymentRecommendations` component itself https://github.com/woocommerce/woocommerce/blob/d4c4c738eb7cec38c557a992102fa4e00ab590ca/plugins/woocommerce/client/admin/client/payments/payment-recommendations.tsx#L214-L214 - it just cares about its own markup/logic